src/Artsy/Relay/createEnvironment.ts

```diff
@@ -10,11 +10,20 @@ import { NetworkError } from "Utils/errors"
 
 import {
   cacheMiddleware,
+  errorMiddleware,
   loggerMiddleware,
   RelayNetworkLayer,
   urlMiddleware,
 } from "react-relay-network-modern"
 
+const isServer = typeof window === "undefined"
+const isDevelopment =
+  (isServer ? process.env.NODE_ENV : sd.NODE_ENV) === "development"
+
+const METAPHYSICS_ENDPOINT = isServer
+  ? process.env.METAPHYSICS_ENDPOINT
+  : sd.METAPHYSICS_ENDPOINT
```

Comment:
Still need to set down a nicer pattern for retrieving client / server values so that these ternaries aren't necessary

Reply:
We already have that, you should only use the value from `sd` https://github.com/artsy/reaction/blob/8f6ac8b73b1c6cb8e79985a77ca86d3f6d06d342/.storybook/webpack.config.js#L33

and we should probably move all other `process.env` occurrences in the codebase to that too.

Reply:
Will also need to figure out how to pipe this sharify config setup through to our tests

Reply:
@alloy - this is a larger task than expected due to tests as well as ensuring that things work properly over in force. Cool if I push this to a separate PR?

Reply:
Oh were you thinking of those cases where server request data is shared to the client via the import? Because global data like these can be considered immutable, they are not set/overwritten on each request.

Reply:
(Hence the separate “Global” and readonly typing of these)

Reply:
Right right, i was just wondering about precident around removing all `process.env` across codebase and replacing with `sd`. What if we're dealing with user data and another dev uses `sd` in an incorrect way, leading to singleton issues? That's what i meant by finding a nice seamless way to handle each case where we don't have to keep these things in mind.

Reply:
Yeah good point, we should have another think when we consider eventually deprecating Force. I think the we way we’ve done it in Reaction (see old LoyaltyApp code) was a good middle ground, as iirc it made sure you couldn’t use the wrong data in the wrong place.